llvm/lib/SandboxIR/Tracker.cpp

```diff
@@ -160,6 +160,34 @@ void CallBrInstSetIndirectDest::dump() const {
 }
 #endif
 
+SetVolatile::SetVolatile(Instruction *I, bool WasBool, Tracker &Tracker)
+    : IRChangeBase(Tracker), Inst(I) {
+  if (auto *Load = cast<llvm::LoadInst>(Inst)) {
+    WasVolatile = Load->isVolatile();
+    InstUnion = Load;
+  }
```

Comment:
We need to have some kind of assertion that this is either a LoadInst or a StoreInst. One way is to early return after `InstUnion = ...` in both this and the following `if` and add an `llvm_unreachable("Expected LoadInst or StoreInst");` at the end of the function.